sagemaker/launch/launch_realtime_endpoint.py

```diff
@@ -0,0 +1,269 @@
+        assert check_tarfile_s3_object(args.model_tarfile_s3) == True, \
+            (f'Not find a tar file in the given S3 URL: {args.model_tarfile_s3}...')
+
+        # model_name is either directly given by users, or had been assigned by the argument
+        # validation function
+        model_name = input_args.model_name
+    else:
+        # prepare task dependent variables
+        entry_point_dir = os.path.join(os.path.dirname(__file__), ENTRY_FOLDER_NAME)
+
+        # TODO: When adding new realtime inference tasks, add new elif here to support them
+        if input_args.infer_task_type == SUPPORTED_REALTIME_INFER_TASKS[0]: # node_classification
+            entrypoint_file_name = list(ENTRY_FILE_NAMES)[0]
+            path_to_entry = os.path.join(entry_point_dir, entrypoint_file_name)
+        else:
+            raise NotImplementedError(f'The given real-time inference task \
+                                        {input_args.infer_task_type} is not supported.')
+
+        path_to_model = os.path.join(input_args.restore_model_path, DEFAULT_GS_MODLE_FILE_NAME)
+        path_to_yaml = input_args.model_yaml_config_file
+        path_to_json = input_args.graph_json_config_file
+        model_name = input_args.model_name
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            model_tarfile_path = wrap_model_artifacts(path_to_model, path_to_yaml, path_to_json,
+                                                    path_to_entry, tmpdirname,
+                                                    output_tarfile_name=model_name)
+            logging.info('Packed and compressed model artifacts into {}.'.format(model_tarfile_path))
+
+            # upload the model tar file to the given S3 bucket
+            model_url_s3 = upload_data_to_s3(input_args.upload_tarfile_s3, model_tarfile_path,
+                                            sm_session)
+            logging.info('Uploaded the model tar file to {}'.format(model_url_s3))
+
+    # ================= create deployable model ================= #
+    image_url = input_args.image_url
+    role = input_args.role
+    sm_client = boto3.client(service_name="sagemaker", region_name=input_args.region)
+
+    container = {
+        "Image": image_url,
+        "ModelDataUrl": model_url_s3,
+        "Environment": {"SAGEMAKER_PROGRAM": entrypoint_file_name}
+    }
+
+    sm_model_name = model_name + '-' + strftime("%Y-%m-%d-%H-%M-%S", gmtime())
+    create_model_response = sm_client.create_model(ModelName=sm_model_name,
+                                                   ExecutionRoleArn=role, Containers=[container])
+    logging.info('Model ARN: {}'.format(create_model_response['ModelArn']))
+
+    # ================= create an endpoint configuration ================= #
+    sm_ep_config_name = model_name + "-EndpointConfig-" + strftime("%Y-%m-%d-%H-%M-%S", gmtime())
+
+    create_endpoint_config_response = sm_client.create_endpoint_config(
```

Comment:
Can users customize the ProductionVariants?

Reply:
Can we make it configurable?

Reply:
Yes, modified to accept a new argument, letting users to configure variants. And then merge them to the values defined in other arguments.